src/Views/Homepage.vala

```diff
@@ -25,44 +25,45 @@ public class AppCenter.Homepage : AbstractView {
     private Gtk.FlowBox category_flow;
     private Gtk.ScrolledWindow category_scrolled;
     private AppStream.Category current_category;
+    private uint banner_timeout_id;
 
     public signal void page_loaded ();
 
     public bool viewing_package { get; private set; default = false; }
 
     public AppStream.Category currently_viewed_category;
 #if HOMEPAGE
-    public Widgets.Banner newest_banner;
-    public Gtk.Revealer switcher_revealer;
-
-    private Widgets.Switcher switcher;
+    private Hdy.Carousel banner_carousel;
+    private Gtk.Revealer banner_revealer;
     private Gtk.FlowBox recently_updated_carousel;
     private Gtk.Revealer recently_updated_revealer;
 
+
     construct {
-        switcher = new Widgets.Switcher () {
-            halign = Gtk.Align.CENTER
+        banner_carousel = new Hdy.Carousel () {
+            allow_long_swipes = true
```

Comment:
Thoughts on this @danrabbit? I'm kind of indifferent but I think you liked enabling long swipes in other contexts.